examples/5_secret_storage.py

```diff
@@ -0,0 +1,419 @@
+#!/usr/bin/env python3
+
+###################################
+# WHAT IS IN THIS EXAMPLE?
+#
+# Keybase has added an encrypted key-value store; see 4_simple_storage.py for
+# more information.
+#
+# This example implements a simple bot to manage hackerspace tool rentals. It
+# shows one way you can obfuscate entryKeys (which are not encrypted) by
+# storing their HMACs, so that no one but your team (not even
+# Keybase) can know about the names of all the cool tools you have; you can do
+# something similar to hide namespaces. Additionally this example handles
+# concurrent writes using a cache to prevent one user from unintentionally
+# clobbering another user's rental updates.
+# ###################################
+
+import asyncio
+import hmac
+import json
+import logging
+import secrets
+import sys
+from base64 import b64decode, b64encode
+from typing import Any, Dict, List, Union
+
+from pykeybasebot import Bot, KVStoreClient
+from pykeybasebot.errors import DeleteNonExistentError, RevisionError
+from pykeybasebot.types import keybase1
+
+logging.basicConfig(level=logging.DEBUG)
+
+if "win32" in sys.platform:
+    # Windows specific event-loop policy
+    asyncio.set_event_loop_policy(
+        asyncio.WindowsProactorEventLoopPolicy()  # type: ignore
+    )
+
+
+class CachedBot(Bot):
+    """
+        Custom bot that has some stateful clients.
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.secret_kvstore_client = SecretKeyKVStoreClient(self)
```

Comment:
this is never referenced from the CachedBot (which is good, because it would be an implicit violation of the law of demeter). i'd suggest make this constraint explicit...
```python
def __init__(self, *args, **kwargs):
    secret_client = SecretKeyKVStoreClient(self)
    self._cached_secret_kvstore = CachedKVStoreClient(self, secret_client)
    super().__init__(*args, **kwargs)

@property
def cached_secret_kvstore(self):
    return self._cached_secret_kvstore
```

if you're feeling bold, you could also make it like this...

```python
@property
def kvstore(self):
    return self._cached_secret_kvstore
```
which might simplify some of the other objects higher up in the stack: i.e. your RentalClient just needs the bot and it can interact with it like it would a normal `bot`, this one has just been juiced up. 

given the way this bot object has developed, i'd also consider renaming it from `CachedBot` to something a little more telling. like maybe `CustomBot` or `JuicedUpBot` or `PrivateConcurrentBot`? dunno. `CachedBot` is a bit misleading.

Reply:
after later comment, adding another suggestion...
```python
class WhateverYouChooseBot(Bot):
    @property
    def kvstore(self):
        if not self._cached_secret_kvstore
            secret_client = SecretKeyKVStoreClient(self)
            self._cached_secret_kvstore = CachedKVStoreClient(secret_client)
        return self._cached_secret_kvstore
```

Reply:
ok i renamed the bot class to CustomKVStoreBot, and this bot has a kvstore property that returns a CachedKVStoreClient.
then i renamed RentalClient to RentalBotClient. and i made RentalBotClient take in a Bot.